File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    sorted_priorities = sorted(priorities, reverse=True)
    q = deque()
    sq = deque(sorted_priorities)
    for i in range(len(priorities)):
        q.append([i,priorities[i]])

    while sq:
        val = sq.popleft()
        while q:
            idx, priority = q.popleft()
            if val == priority:
                answer+=1
                break
            else:
                q.append([idx, priority])
        if location == idx:
            break
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let q = priorities.map((priority, idx) => ({idx, priority}));
    let answer = 0;
    
    while (q.length > 0) {
        const current = q.shift();
        if (q.some(item => item.priority > current.priority)) {
            q.push(current)
        } else {
            answer++;
            if (current.idx === location) {
                return answer
            }
            
        }
    }
}